src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRangeHandler.cs

```diff
@@ -117,13 +130,32 @@ public void Dispose()
         }
 
         private void OnLspSolutionChanged(object? sender, WorkspaceChangeEventArgs e)
-            => EnqueueSemanticTokenRefreshNotification();
+        {
+            if (e.DocumentId is not null && e.Kind is WorkspaceChangeKind.DocumentChanged)
+            {
+                var document = e.NewSolution.GetDocument(e.DocumentId);
+                if (document is not null)
```

Comment:
Can document ever be null here? If so, should we enqueue `null` refresh notification just in case?

perhaps `if (e.Document is not null && e.Kind is DocumentChanged && e.NewSolution.GetDocument(e.DocumentId) is { } document)` would be a better check in the if statement above?

Reply:
DavidB made a comment that technically shouldn't be null so I've removed the check, but that's good syntax to keep in mind for next time, thanks!